spartan/aztec-bot/values.yaml

```
@@ -28,6 +28,51 @@ bot:
 
   replicaCount: 1
 
+  initContainers:
+    - name: wait-for-block-production
+      image: curlimages/curl:latest
+      command:
+        - /bin/sh
+        - -c
+        - |
+          AZTEC_NODE_URL="${AZTEC_NODE_URL}"
+          if [ -z "$AZTEC_NODE_URL" ]; then
+            echo "AZTEC_NODE_URL not set, skipping block production wait"
+            exit 0
+          fi
+
+          echo "Waiting for Aztec Node to be available at ${AZTEC_NODE_URL}..."
+          until curl -s ${AZTEC_NODE_URL}/status > /dev/null; do
+            echo "Waiting for aztec-node..."
+            sleep 2
+          done
+          echo "Aztec Node is ready"
+
+          echo "Waiting for block production to start..."
+          TIMEOUT=300
+          ELAPSED=0
+          while [ $ELAPSED -lt $TIMEOUT ]; do
+            BLOCK_NUMBER=$(curl -s -X POST -H 'Content-Type: application/json' \
+              -d '{"jsonrpc":"2.0","method":"node_getBlockNumber","params":[],"id":1}' \
+              ${AZTEC_NODE_URL} | grep -o '"result":[0-9]*' | cut -d':' -f2)
```

if `jq` is available in `curlimages/curl` this makes the script simpler

```suggestion
            BLOCK_NUMBER=$(curl -s -X POST -H 'Content-Type: application/json' \
              -d '{"jsonrpc":"2.0","method":"node_getBlockNumber","params":[],"id":1}' \
              ${AZTEC_NODE_URL} | jq -r ".result")
```

---

from quick check, doesn't look like it is